src/home/homework-luks.c

```diff
@@ -3214,9 +3259,13 @@ int home_resize_luks(
                         new_partition_size = 0;
                         intention = INTENTION_SHRINK;
                 } else {
-                        uint64_t new_partition_size_rounded;
+                        uint64_t new_partition_size_rounded = DISK_SIZE_ROUND_DOWN(h->disk_size);
 
-                        new_partition_size_rounded = DISK_SIZE_ROUND_DOWN(h->disk_size);
+                        if (h->disk_size == UINT64_MAX && partition != NULL) {
```

Comment:
our usual coding style is that we check pointers for non-NULL relying on C's downgrade-to-bool feature. i.e. please change to:

```c
if (h->disk_size == UINT64_MAX && !partition) {
```

but doesn't matter much

Reply:
I changed it to:
```C
if (h->disk_size == UINT64_MAX && partition) {
```